src/persist-txn/src/txns.rs

```diff
@@ -568,6 +576,28 @@ where
                 })
                 .collect();
 
+            // TODO(jkosh44) This is technically correct, but it might be better to combine this
+            // with the for loop above and go through everything in ts order.
+            while let Some((_, register_ts)) = self.txns_cache.unapplied_registers.front() {
+                if ts < register_ts {
+                    break;
+                }
+                // TODO(jkosh44) Is this the right place to remove elements? They're fully applied
```

Comment:
Hmm, that's a great question and reasonably subtle. I think either is workable. Seems like the inputs boil down to:

In apply:
- Tidy is currently just things to write down durably and this isn't that
- Takes effect immediately, so if we get a compact_to call next, we'll be able to downgrade the maximum possible amount

In tidy:
- Apply currently doesn't alter the cache directly, so marking as batch apply as done vs a register would happen in different places

Given the above, I lean toward what you have. If anything, it's maybe an argument that we could do the same here for batches (though we'd have to account for it when we do actually read the retraction back in via the Subscribe, so I def wouldn't do it in this PR)